server/handler/base.go

```diff
@@ -65,19 +73,40 @@ func (p *PullEvaluationOptions) FillDefaults() {
 	}
 }
 
-func (b *Base) MakeStatus(state, message string, pr *github.PullRequest) *github.RepoStatus {
-	var details string
+func (b *Base) PostStatus(ctx context.Context, client *github.Client, owner, repo, ref string, state, message string, pr *github.PullRequest) error {
+	var detailsURL string
 	if pr != nil {
 		publicURL := strings.TrimSuffix(b.BaseConfig.PublicURL, "/")
-		details = fmt.Sprintf("%s/details/%s/%d", publicURL, pr.GetBase().GetRepo().GetFullName(), pr.GetNumber())
+		detailsURL = fmt.Sprintf("%s/details/%s/%d", publicURL, pr.GetBase().GetRepo().GetFullName(), pr.GetNumber())
 	}
 
-	return &github.RepoStatus{
-		Context:     &b.PullOpts.StatusCheckContext,
+	contextWithBranch := fmt.Sprintf("%s: %s", b.PullOpts.StatusCheckContext, pr.GetBase().GetLabel())
+	status := &github.RepoStatus{
+		Context:     &contextWithBranch,
 		State:       &state,
 		Description: &message,
-		TargetURL:   &details,
+		TargetURL:   &detailsURL,
+	}
+
+	if err := b.postGitHubRepoStatus(ctx, client, status, owner, repo, ref); err != nil {
+		return err
+	}
+
+	if b.PullOpts.PostInsecureStatusChecks {
+		status.Context = &b.PullOpts.StatusCheckContext
+		if err := b.postGitHubRepoStatus(ctx, client, status, owner, repo, ref); err != nil {
+			return err
+		}
 	}
+
+	return nil
+}
+
+func (b *Base) postGitHubRepoStatus(ctx context.Context, client *github.Client, status *github.RepoStatus, owner, repo, ref string) error {
```

Comment:
style: I think it's slightly more idiomatic to take the `status` as the last argument